coresimd/powerpc/altivec.rs

```diff
@@ -355,10 +358,53 @@ impl_from_bits_!(
     vector_bool_int
 );
 
+#[allow(improper_ctypes)]
+extern "C" {
+#[ link_name = "llvm.ppc.altivec.vperm" ]
+fn vperm(a: vector_signed_int, b: vector_signed_int, c: vector_unsigned_char) -> vector_signed_int;
+}
+
 mod sealed {
 
     use super::*;
 
+    #[inline]
+    #[target_feature(enable = "altivec")]
+    #[cfg_attr(test, assert_instr(vperm))]
+    unsafe fn vec_vperm(a: vector_signed_int, b: vector_signed_int, c: vector_unsigned_char) -> vector_signed_int {
```

Comment:
I think we should add `assert_instr` tests for all vector types.

Reply:
That function is the only one that is called for every vector type.

All the vector types are transmuted to i32x4.